### PR TITLE
Add Text Shortcuts extension

### PR DESCRIPTION
**Pull request by koinzhang**
## Description

**Main Features**

A powerful and easy to use tool that will help you transform your text.
Text Shortcuts removes the need to perform repetitive tasks manually on your text.
Combine transformation actions to create powerful automations for your text.

Built-in multiple shortcuts to support installation ready to use!

**Command:**

- **Create Text Shortcut**: Create Text Shortcut with actions
- **Search Text Shortcut**: Search for built-in or custom shortcuts to handle selected or clipboard text
- **Run Text Shortcut Now**: Run actions directly to handle selected or clipboard text

## Screencast

https://user-images.githubusercontent.com/36128970/158920169-43c9ac59-a285-4dbc-b317-2727d856a394.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `metadata` folder

### CODE REVIEW

File: extensions/text-shortcuts/src/search-shortcut.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    _fetchBuildInShortcut().then();
+  }, [updateList]);
+
+  return (
+    <List
+      isShowingDetail={preferences().detail}
+      isLoading={allShortcuts.length == 0}
+      searchBarPlaceholder={"Search shortcut"}
+      onSelectionChange={async (id) => {
+        const _inputItem = await fetchItemInput();
```
Comment (sxn) [inline extensions/text-shortcuts/src/search-shortcut.tsx:74]: hi @koinzhang - it seems like it's not very useful to call this function in the selection change handler. What about reading the data from the frontmost app/clipboard only once?

Comment (sxn) [reply extensions/text-shortcuts/src/search-shortcut.tsx:74]: hi @koinzhang - thanks for the explanation 🤔 let me know when you've had a chance to fix the logic in `fetchItemInput`.

Regarding the feature request: best to open an [issue](https://github.com/raycast/extensions/issues/new?assignees=&labels=api%2C+feature+request&template=api_feature_request.md&title=) so we can better discuss the use cases & track progress. 😺

Comment (koinzhang) [reply extensions/text-shortcuts/src/search-shortcut.tsx]: Hi, @sxn. I have updated the code, please re-review it.
I'll raise an issue to discuss about the suggestion mentioned, thanks for reminding!

File: extensions/text-shortcuts/src/create-shortcut.tsx [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+        title="More Actions"
+        icon={Icon.Terminal}
+        onAction={async () => {
+          try {
+            await simulatePressKeyboard(40, "command");
+          } catch (e) {
+            await showToast(Toast.Style.Failure, String(e));
+          }
+        }}
+      />
```
Comment (sxn) [inline extensions/text-shortcuts/src/create-shortcut.tsx:284]:
```suggestion
```

Raycast already has a global for opening the action panel (`⌘K`), so creating an action for it is superfluous. Additionally, making it he primary action can mess with user expectation: I'd expect the primary action on a `Form` to…submit the form. 😸

Comment (koinzhang) [reply extensions/text-shortcuts/src/create-shortcut.tsx:284]: Okay, here I was mainly worried that the user would create a shortcut directly without adding an action, so I wrote it this way.
But now I've added help text and a hint to detect if the action is included, so I'll change it here

**Comment by sxn** [comment]
as discussed on Slack, I've updated a couple of typos in the app & updated the command names a bit.

Ping me once you've had a chance to look into the issue with the `NSBeep` sound being produced every time the selection changes. 😸

**Comment by sxn** [comment]
🎉

**Pull request merged**